pytorch_lightning/profiler/pytorch.py

```diff
@@ -607,40 +628,26 @@ def __init__(
             self.context_names = {}
             self.running_stack = []
             self.profiler = None
-            self.lightning_module = None
+            self.lightning_module = None  # TODO: where is this set?
             self.register = None
 
             self.output_fname = output_filename
             self.output_file = None
             if local_rank is not None:
-                super().on_train_start(local_rank=local_rank)
+                super().on_train_start(local_rank=local_rank)  # TODO: why super()?
                 self.on_train_start = super().on_train_start
 
-        def summary(self) -> str:
-            recorded_stats = {}
-            output_string = ''
-            local_rank = '0' if self.local_rank is None else self.local_rank
-
-            if not self.enabled:
-                return output_string
-
+        def _summary(self, local_rank: int) -> dict:
             self.profiler.__exit__(None, None, None)
             if self.register is not None:
                 self.register.__exit__(None, None, None)
 
             data = self.profiler.events().key_averages(group_by_input_shapes=self.group_by_input_shapes)
             table = data.table(sort_by=self.sort_by_key, row_limit=self.row_limit)
-            recorded_stats[self.START_ACTION] = table
+            return {self.START_ACTION: table}
```

Comment:
Does it make sense that this returns just one action?

Reply:
It is being logged in the terminal. This action covers the entire training / validation.

Reply:
So if this supported other calls like `.test` then would it have more actions?

Reply:
Currently doesn't but yes it should.